include/odp/api/spec/packet.h

```diff
@@ -1799,15 +1799,23 @@ uint32_t odp_packet_flow_hash(odp_packet_t pkt);
  * Store the packet flow hash for the packet and sets the flow hash flag. This
  * enables (but does not require!) application to reflect packet header
  * changes in the hash.
+ * When scheduler is configured as flow aware, schedule queue synchronization
+ * will be based on flow within each queue.
+ * When scheduler is configured as flow unaware, packet flow hash is ignored by
+ * the implementation.
+ * The value of flow hash should not be greater than the max flow count
+ * supported by the implementation.
  *
  * @param      pkt              Packet handle
  * @param      flow_hash        Hash value to set
  *
  * @note If the platform needs to keep the original hash value, it has to
- * maintain it internally. Overwriting the platform provided value doesn't
- * change how the platform handles this packet after it.
+ * maintain it internally.
+ * When scheduler is configured as flow unaware, overwriting the platform
+ * provided value doesn't change how the platform handles this packet after it.
```

Comment:
I'd document this the other way (positive vs. negative spec text):
" When scheduler is configured as flow aware, the flow hash value affects scheduling of the event."

Reply:
Actually, we'll need this flow ID for all events, not just packets as it affects event synchronization. E.g. application may need to sync timeouts / ctrl plane messages with packets of a flow. So, application would need to set the same flow ID for those events as well.

odp_event_flow_id() / odp_event_flow_id_set() are needed. Maybe only the event level function is needed, as scheduler and queue enq/deq work with events anyway.

Reply:
This flow_hash() could keep it's original purpose as input HW calculated 32 bit hash that does not affect scheduling. Only the event level flow ID would affect scheduling. Those could contain the same value for incoming packets, or event flow ID would have only e.g. 20bits but packet flow hash full 32 bits.